### PR TITLE
Filipposl/coingecko plugin

### PR DESCRIPTION
**Pull request by filipposl**
# Background

## What does this PR do?
Introduces a new Coingecko plugin that enables querying cryptocurrency market data directly through the GOAT SDK. 

The PR includes:
* New Coingecko plugin implementation
* Vercel AI example demonstrating plugin usage
* Integration with existing SDK architecture
* Support for querying trending cryptocurrencies and price data for specified listing

## What kind of change is this?
New feature

# Documentation changes needed?
If a docs change is needed: I have updated the documentation accordingly.

# Testing

## Detailed testing steps
1. Action: Install plugin and configure with Coingecko API key
> Expected result: The plugin initializes without errors

2. Action: Query price for a specific cryptocurrency (e.g., "BONK")
> Expected result: Returns current price data in specified currency

3. Action: Request trending cryptocurrencies list
> Expected result: Returns array of currently trending cryptocurrencies

4. Action: Run the example script with a natural language query
> Expected result: AI correctly interprets the query and returns relevant cryptocurrency data

## For plugins
- [*] I have tested this change locally with key pair wallets
- [ ] I have tested this change locally with hosted wallets (e.g. Crossmint Smart Wallets, etc.)

## Discord username
flympe

### CODE REVIEW

File: typescript/packages/plugins/coingecko/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Usage
+    
+```typescript
+import { coingecko } from "@goat-sdk/plugin-coingecko";
+
+const plugin = coingecko({ key: process.env.COINGECKO_API_KEY as string });
+
+// Get trending coins
+const trendingCoins = await plugin.getTrendingCoins();
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/README.md:18]: The plugin returns an array so you can't really really call the function like this.

For the readme I would simply add how you get the plugin initialized and describe the actions it has

File: typescript/packages/plugins/coingecko/src/tools.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    });
+    const response = await fetch(`https://api.coingecko.com/api/v3/simple/price?${params.toString()}`);
+    if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    return await response.json();
+}
+
+export function getTools(credentials: {
+	key: string;
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/src/tools.ts:31]: Key is not being used, do we need it?

File: typescript/packages/plugins/coingecko/src/tools.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        parameters: getTrendingCoinsParametersSchema,
+        method: async () => fetchTrendingCoins(),
+    };
+
+    const getCoinPriceTool: DeferredTool<any> = {
+        name: 'get_coin_price',
+        description: 'This tool fetches the price of a specific coin from CoinGecko',
+        parameters: getCoinPriceParametersSchema,
+        method: async (
+            _client: any,
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/src/tools.ts:47]: The lint will complain about any. Here it's WalletClient

File: typescript/packages/plugins/coingecko/src/parameters.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	limit: z.number().optional().describe("The number of trending coins to return. Defaults to all coins."),
+	include_platform: z.boolean().optional().describe("Include platform contract addresses (e.g., ETH, BSC) in response"),
+});
+
+export const getCoinPriceParametersSchema = z.object({
+	coinId: z.string().describe("The ID of the coin on CoinGecko (e.g., 'bitcoin', 'ethereum')"),
+	vsCurrency: z.string()
+		.default("usd")
+		.describe("The target currency to get price in (e.g., 'usd', 'eur', 'jpy')"),
+	include_market_cap: z.boolean()
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/src/parameters.ts:13]: These parameters are not being used. Do we need them? Also I would keep them consistent making them camel case

File: typescript/packages/plugins/coingecko/src/tools.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export function getTools(credentials: {
+	key: string;
+}): DeferredTool<any>[] {
+    const tools: DeferredTool<any>[] = [];
+
+    const getTrendingCoinsTool: DeferredTool<any> = {
+        name: 'get_trending_coins',
+        description: 'This tool fetches the list of trending coins from CoinGecko',
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/src/tools.ts:37]: Check out the other plugins, use this `{{tool}}` 

Reason explained in the step 2 of the plugin tutorial https://ohmygoat.dev/plugins

File: typescript/packages/plugins/coingecko/src/tools.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    return await response.json();
+}
+
+export function getTools(credentials: {
+	key: string;
+}): DeferredTool<any>[] {
+    const tools: DeferredTool<any>[] = [];
+
+    const getTrendingCoinsTool: DeferredTool<any> = {
+        name: 'get_trending_coins',
```
Comment (aigustin) [inline typescript/packages/plugins/coingecko/src/tools.ts:36]: Should we call them coins? or tokens?

Comment (aigustin) [reply typescript/packages/plugins/coingecko/src/tools.ts:36]: I guess coin is better since its the language of the api

**Review by aigustin — APPROVED** [review]
Very cool!! Made some comments! But otherwise lgtm!

File: typescript/packages/plugins/coingecko/src/tools.ts (hunk 2 above)
Comment (filipposl) [reply typescript/packages/plugins/coingecko/src/tools.ts:31]: It's passed to the api calls the tools use

File: typescript/examples/vercel-ai/coingecko/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	},
+	"author": "",
+	"license": "MIT",
+	"dependencies": {
+		"@ai-sdk/openai": "^1.0.4",
+		"@goat-sdk/adapter-vercel-ai": "0.1.4",
+		"@goat-sdk/core": "0.3.8",
+		"@goat-sdk/plugin-coingecko": "workspace:*",
+		"@goat-sdk/plugin-erc20": "0.1.4", 
+		"@goat-sdk/wallet-viem": "0.1.3",
```
Comment (mPaella) [inline typescript/examples/vercel-ai/coingecko/package.json:17]: `workspace:*` for all of these

**Review by 0xaguspunk — APPROVED** [review]

**Comment by sachiew** [comment]
hey @filipposl , thanks for contributing this. I work on coinGecko product, would you be keen to add more support here for CoinGecko plugin. There's a lot more can be done with CG public and pro api. Happy to bounce ideas, and grant you paid api keys for testing & personal use.  Feel free to ping TG @soonegg91

ICYMI: CG API now supports onchain data, with 100+ networks
https://docs.coingecko.com/reference/endpoint-overview#-onchain-dex-endpoints-geckoterminal

**Comment by filipposl** [comment]
for sure - will get in touch via TG!

**Pull request merged**